src/main/java/org/fcrepo/auth/webac/WebACRolesProvider.java

```diff
@@ -98,8 +99,20 @@ public void deleteRoles(final Node node) throws RepositoryException {
     @Override
     public Map<String, List<String>> findRolesForPath(final Path absPath, final Session session)
             throws RepositoryException {
-        LOGGER.debug("findRolesForPath: {}", absPath.toString());
-        return getAgentRoles(nodeService.find(session, absPath.toString()));
+        return getAgentRoles(locateResource(absPath, session));
+    }
+
+    private FedoraResource locateResource(final Path path, final Session session) {
+
+        final Predicate<Path> exists = UncheckedPredicate.uncheck(x -> session.nodeExists(x.toString()));
+
+        if (exists.test(path) || path.isRoot()) {
+            LOGGER.debug("findRolesForPath: {}", path.getString());
+            return nodeService.find(session, path.toString());
+        } else {
```

Comment:
That is what I mean... and I am looking out for @ajs6f here, as he seems to have a strong preference.